spec/rubocop/cop/style/redundant_return_spec.rb

```diff
@@ -54,6 +54,40 @@ def func
     RUBY
   end
 
+  it 'reports an offense for define_method ending with return' do
+    expect_offense(<<~RUBY)
+      define_method(:foo) do
+        some_preceding_statements
+        return something
+        ^^^^^^ Redundant `return` detected.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      define_method(:foo) do
+        some_preceding_statements
+        something
+      end
+    RUBY
+  end
```

Comment:
The following false negative and autocorrection should be resolved.

```ruby
    expect_offense(<<~RUBY)
      define_method(:foo) do
        return something
        ^^^^^^ Redundant `return` detected.
      end
    RUBY
```

Reply:
Thank you very much. I updated this PR.